back-end/src/base/calls/calls.repository.ts

```diff
@@ -145,8 +170,70 @@ export class CallsRepository {
 
 		return classroom.students;
 	}
-	async getStudentList(studentIdList: Array<ObjectId>) {
-		const studentList = await this.db.collection('users').find({ _id: { $in: studentIdList } }).toArray();
+	async getStudentList(courseId: string, studentIdList: Array<ObjectId>) {
+		const studentList = await this.db
+			.collection('users')
+			.find({ _id: { $in: studentIdList } })
+			.toArray();
+
+		// Add the student's presence to the student object
+		for (let i = 0; i < studentList.length; i++) {
+			const student = studentList[i];
+			const studentId = student._id.toString();
+			const presence = await this.getStudentPresence(courseId, studentId);
+			studentList[i] = { ...student, presence: presence };
+		}
 		return studentList;
 	}
+
+	async getStudentPresence(courseId: string, studentId: string) {
+		const courseObjectId = new ObjectId(courseId);
+		const period = ['arrival', 'departure'];
+		const date = new Date();
+		const periodIndex = date.getHours() < 16 ? 0 : 1;
+		let call = null;
+		try {
+			call = await this.db
```

Comment:
Honestly, it's very rare that a mongo query fail.
One case it could happen is if you try to update a document and the $set isn't present. Also, if you wish to log error, use Logger from "nest" instead.